dashboard/client/src/common/MultiTabLogViewer.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Box,
+  createStyles,
+  IconButton,
+  makeStyles,
+  Tab,
+  Tabs,
+  Typography,
+} from "@material-ui/core";
+import React, { useState } from "react";
+import { RiArrowUpDownLine } from "react-icons/ri";
+import { useStateApiLogs } from "../pages/log/hooks";
+import { LogViewer } from "../pages/log/LogViewer";
+import { HideableBlock } from "./CollapsibleSection";
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    tabs: {
+      borderBottom: `1px solid ${theme.palette.divider}`,
+    },
+  }),
+);
+
+export type MultiTabLogViewerTab = {
+  title: string;
+  nodeId: string | null;
+  filename?: string;
+};
+
+export type MultiTabLogViewerProps = {
+  tabs: MultiTabLogViewerTab[];
+};
+
+export const MultiTabLogViewer = ({ tabs }: MultiTabLogViewerProps) => {
+  const classes = useStyles();
+  const [value, setValue] = useState(tabs[0]?.title);
```

Comment:
nit: you could have some type such as 
```
type NonEmptyArray<T> = [T, ...T[]];
```
and declare tabs field as `NonEmptyArray<MultiTabLogViewerTab>`

it would then both do a type check and remove the optional chaining `?` here

Reply:
Actually an empty array is supported in a future PR. We render a special message in that case.